Atomic/trust.py

```diff
@@ -351,26 +352,73 @@ def show(self):
-                                table[key] = { "type": value[0]["type"] }
+                            for key, values in policy["transports"][tt].items():
+                                table[key] = { "type": values[0]["type"] }
+                                table[key]["keys"] = []
+                                for v in values:
+                                    if "keyPath" in v:
+                                        table[key]["keys"].append(v["keyPath"])
+                                    elif "keyData" in v:
+                                        table[key]["keys"].append(v["keyData"])
                                 reg_info = util.have_match_registry(key, registry_configs)
                                 if reg_info:
                                     table[key]["sigstore"] = reg_info["sigstore"]
                                 else:
                                     table[key]["sigstore"] = ""
-
+                if "default" in policy:
+                    table["* (default)"] = { "type": policy["default"][0]["type"], "keys": None, "sigstore": "" }
                 sorted_table = collections.OrderedDict(sorted(table.items()))
-                for key, value in sorted_table.items():
-                    util.write_out('{0:<35} {1:<8} {2}'.format(key, self.trusttype_map(value["type"]), value["sigstore"]))
-                util.write_out('{0:<35} {1:<8}'.format("* (default)", self.trusttype_map(policy["default"][0]["type"])))
-        return policy
+                if self.args.json:
+                    util.write_out(json.dumps(sorted_table, indent=4))
+                    return
+                else:
+                    for key, value in sorted_table.items():
+                        util.write_out('{0:<35} {1:<6} {2:<29} {3}'.format(key, self.trusttype_map(value["type"]), self.get_gpg_id(value["keys"]), value["sigstore"]))
+                    #util.write_out('{0:<35} {1:<8}'.format("* (default)", self.trusttype_map(policy["default"][0]["type"])))
+        #return policy
+
+    def get_gpg_id(self, keys):
+        """
+        Return GPG identity, either bracketed <email> or ID string
+        comma separated if more than one key
+        :param keys: list of gpg key file paths (keyPath) and/or inline key payload (keyData)
+        :return: key id string or empty string
+        """
+        if not keys:
+            return ""
+        keylist = None
+        for key in keys:
+            if not os.path.exists(key):
+                # extend when we support parsing inline 'keyData' base64 encoded key
+                return ""
+            else:
+                cmd = ["gpg2", "--list-packets", key]
+            try:
+                results = util.check_output(cmd)
+            except util.FileNotFound:
+                results = ""
+            lines = results.split(b'\n')
+            for line in lines:
+                line = str(line)
+                if "user ID packet" in line:
```

Comment:
After some [research](http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob_plain;f=doc/DETAILS) it looks like the intended machine parsing output is to use `--with-colons` for a stable, colon-delimited output.